src/metabase/analyze/fingerprint/insights.clj

```diff
@@ -196,9 +196,8 @@
 ;; We downsize UNIX timestamps to lessen the chance of overflows and numerical instabilities.
 (def ^Long ^:const ^:private ms-in-a-day (* 1000 60 60 24))
 
-(defn- ms->day
-  [dt]
-  (/ dt ms-in-a-day))
+(defn- ms->day [dt]
+  (quot dt ms-in-a-day))
```

Comment:
What are the consequences of the truncation? It seems more prudent to return a float instead.

Reply:
Honestly, no idea. If the plan was to avoid overflows, than casting milliseconds as double should have been enough.  But you have a point, returning double at least preservers the current behavior, whatever it might intend.